From ca172a715e30cb73a595ccec078b8757aed78f87 Mon Sep 17 00:00:00 2001
Message-Id: <ca172a715e30cb73a595ccec078b8757aed78f87.1440106296.git.chang-joon.lee@intel.com>
In-Reply-To: <1f426693c14c79e65ad8e62ef6bb4b07d6484040.1440106296.git.chang-joon.lee@intel.com>
References: <1f426693c14c79e65ad8e62ef6bb4b07d6484040.1440106296.git.chang-joon.lee@intel.com>
From: Tim Gore <tim.gore@intel.com>
Date: Thu, 30 Jul 2015 10:35:57 +0100
Subject: [PATCH 05/11] Revert "FOR_UPSTREAM [VPG]: drm/i915: 2nd fix for reset
 stats"

This reverts commit 181197ebd5d7701379a0b40f12c52a48d2c6a6d1.

The original commit was to aid the OGL robustness interface
in detecting per engine reset when that reset had not
actually affected the context that it was considering.
After some discussion it was decided to revert this commit
and add code in the usermode OGL driver to look at
batch_pending and batch_active counts across all relevant
contexts to determine the impact of per engine resets.
This is almost certainly a better solution.

However, some igt gem_reset_stats subtests were fixed
by this patch, so now they are broken again when TDR is
enabled. There are several subtests that explicitly check
the reset count returned by i915_get_reset_stats_ioctl and
these now fail when TDR is enabled because TDR does not do
a full GPU reset and so does not increment the reset count.
When TDR goes upstream we will need to make the
gem_reset_stats test TDR aware but for now we probably
need to just ignore these subtests.

Change-Id: I05456cfac607bfed4dfd744480d49a1e42fdbf50
Tracked-On: https://jira01.devtools.intel.com/browse/VIZ-5904
Signed-off-by: Tim Gore <tim.gore@intel.com>

## Don't forget to add a JIRA URL to the Tracked-On metadata!
## https://wiki.ith.intel.com/display/CACTUS/Patch+requirements+check
## url should be in the form:
## Tracked-On: https://jira01.devtools.intel.com/browse/<PROJ>-<NUMB>
## Several Tracked-On lines are allowed
## you can fully prefill your Tracked-On commits using environment variable:
## export CURRENT_TRACKED_ON=https://jira01.devtools.intel.com/browse/<PROJ>-<NUMB>
##
---
 drivers/gpu/drm/i915/intel_uncore.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_uncore.c b/drivers/gpu/drm/i915/intel_uncore.c
index 9c32573..7d96124 100644
--- a/drivers/gpu/drm/i915/intel_uncore.c
+++ b/drivers/gpu/drm/i915/intel_uncore.c
@@ -1084,12 +1084,9 @@ int i915_get_reset_stats_ioctl(struct drm_device *dev,
 	}
 	hs = &ctx->hang_stats;
 
-	if (capable(CAP_SYS_ADMIN)) {
-		int i;
+	if (capable(CAP_SYS_ADMIN))
 		args->reset_count = i915_reset_count(&dev_priv->gpu_error);
-		for (i = 0; i < I915_NUM_RINGS; ++i)
-			args->reset_count += dev_priv->ring[i].hangcheck.total;
-	} else
+	else
 		args->reset_count = 0;
 
 	args->batch_active = hs->batch_active;
-- 
1.9.1

